### PR TITLE
`updateGlobalAssemblyNum` called twice during `loadState`

### PR DESCRIPTION
**Pull request by keckler**
## Description

In response to #690 .

The duplicate call has been removed and another input option `updateGlobalAssemNum` has been added to `loadState` so that it can be passed along to `load`. This way both calls are consistent and `updateGlobalAssemblyNum` doesn't need to be called within `loadState` itself.

The default value of `updateGlobalAssemNum` is `True` so that the previous behavior is maintained by default.

Tests have been added as well.

---

## Checklist

- [X] The code is understandable and maintainable to people beyond the author.
- [X] Tests have been added/updated to verify that the new or changed code works.
- [X] There is no commented out code in this PR.
- [X] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [X] All docstrings are still up-to-date with these changes.

### CODE REVIEW

File: armi/bookkeeping/db/database3.py [hunk 1]
```diff
@@ -439,11 +441,6 @@ def loadState(
                 )
             )
 
-        if updateGlobalAssemNum:
-            updateGlobalAssemblyNum(newR)
-
-        self.o.reattach(newR, self.cs)
```
Comment (john-science) [inline armi/bookkeeping/db/database3.py:445]: Why did you remove the `self.o.reattach()`?

Just curious.

Comment (keckler) [reply armi/bookkeeping/db/database3.py:445]: I didn't remove it, I just moved it closer to where it is actually relevant.

The `else` clause of that conditional only raises errors. So having the `reattach` wait until after the `else` didn't seem necessary, and I just thought it helped with code clarity to move the call into the `if` clause.

**Review by john-science — APPROVED** [review]
